config/main.py

```diff
@@ -2064,6 +2064,538 @@ def delete_snmptrap_server(ctx, ver):
+            raise click.Abort()
+
+
+@community.command('del')
+@click.argument('community', metavar='<snmp_community>', required=True)
+@clicommon.pass_db
+def del_community(db, community):
+    """ Delete snmp community string"""
+    snmp_communities = db.cfgdb.get_table("SNMP_COMMUNITY")
+    if community not in snmp_communities:
+        click.echo("SNMP community {} is not configured".format(community))
+        sys.exit(1)
+    else:
+        db.cfgdb.set_entry('SNMP_COMMUNITY', community, None)
+        click.echo("SNMP community {} removed from configuration".format(community))
+        try:
+            click.echo("Restarting SNMP service...")
+            clicommon.run_command("systemctl reset-failed snmp.service", display_cmd=False)
+            clicommon.run_command("systemctl restart snmp.service", display_cmd=False)
+        except SystemExit as e:
+            click.echo("Restart service snmp failed with error {}".format(e))
+            raise click.Abort()
+
+
+@community.command('replace')
+@click.argument('current_community', metavar='<current_community_string>', required=True)
+@click.argument('new_community', metavar='<new_community_string>', required=True)
+@clicommon.pass_db
+def replace_community(db, current_community, new_community):
+    """ Replace snmp community string"""
+    snmp_communities = db.cfgdb.get_table("SNMP_COMMUNITY")
+    if not current_community in snmp_communities:
+        click.echo("Current SNMP community {} is not configured".format(current_community))
+        sys.exit(1)
+    if not snmp_community_secret_check(new_community):
+        click.echo("Bad community")
+        sys.exit(2)
+    elif new_community in snmp_communities:
+        click.echo("New SNMP community {} to replace current SNMP community {} already "
+                   "configured".format(new_community, current_community))
+        sys.exit(3)
+    else:
+        string_type = snmp_communities[current_community]['TYPE']
+        db.cfgdb.set_entry('SNMP_COMMUNITY', new_community, {'TYPE': string_type})
+        click.echo("SNMP community {} added to configuration".format(new_community))
+        db.cfgdb.set_entry('SNMP_COMMUNITY', current_community, None)
+        click.echo('SNMP community {} removed from configuration'.format(current_community))
+        click.echo('SNMP community {} replace community {}'.format(new_community, current_community))
+        try:
+            click.echo("Restarting SNMP service...")
+            clicommon.run_command("systemctl reset-failed snmp.service", display_cmd=False)
+            clicommon.run_command("systemctl restart snmp.service", display_cmd=False)
+        except SystemExit as e:
+            click.echo("Restart service snmp failed with error {}".format(e))
+            raise click.Abort()
+
+
+@snmp.group(cls=clicommon.AbbreviationGroup)
+@clicommon.pass_db
+def contact(db):
+    pass
+
+
+def is_valid_email(email):
```

Comment:
>is_valid_email [](http://example.com/codeflow?start=4&length=14)

Could you write some unit testcases for this function?

Reply:
Added some additional tests for this.

Reply:
I see you added one test case `test_config_snmp_contact_add_invalid_email`. Actually you can test this function directly `is_valid_email` instead of using `CliRunner`.

Suggest add more test cases to cover as many as postive/negative for this regex.

Reply:
added additional tests.